### ISSUE TITLE
Unhandled exception for Gitlab comment reporter

### ISSUE THREAD

**Issue by 1vm0** [megalinter/megalinter#1246]
If the environment variable `GITLAB_COMMENT_REPORTER` set as `true`, but the pipeline being executed is not a merge request pipeline, then the python-gitlab package will throw a `gitlab.exceptions.GitlabParsingError` exception and GitlabCommentReporter does not catch it and megalinter will crash. 

traceback:
```python
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.9/site-packages/megalinter-0.1-py3.9.egg/megalinter/run.py", line 15, in <module>
    linter.run()
  File "/usr/local/lib/python3.9/site-packages/megalinter-0.1-py3.9.egg/megalinter/MegaLinter.py", line 172, in run
    reporter.produce_report()
  File "/usr/local/lib/python3.9/site-packages/megalinter-0.1-py3.9.egg/megalinter/reporters/GitlabCommentReporter.py", line 72, in produce_report
    mr = project.mergerequests.get(gitlab_merge_request_id)
  File "/usr/local/lib/python3.9/site-packages/python_gitlab-3.1.1-py3.9.egg/gitlab/v4/objects/merge_requests.py", line 467, in get
    return cast(ProjectMergeRequest, super().get(id=id, lazy=lazy, **kwargs))
  File "/usr/local/lib/python3.9/site-packages/python_gitlab-3.1.1-py3.9.egg/gitlab/exceptions.py", line 311, in wrapped_f
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/python_gitlab-3.1.1-py3.9.egg/gitlab/mixins.py", line 114, in get
    return self._obj_cls(self, server_data)
  File "/usr/local/lib/python3.9/site-packages/python_gitlab-3.1.1-py3.9.egg/gitlab/base.py", line 73, in __init__
    raise GitlabParsingError(
```

This happens because if the CI_MERGE_REQUEST_ID and CI_OPEN_MERGE_REQUESTS environment variables are missing, then the `gitlab_merge_request_id` variable is set to empty string.
